File: CLASE VII - Listas Void con Struct Anidadas/pre_parcial/c/destino.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "destino.h"

struct Destino{
    char ciudad[20];
    float latitud;
    float longitud;
};


DestinoPtr crearDestino(char* ciudad, float latitud, float longitud){
    DestinoPtr d = (DestinoPtr) malloc(sizeof(struct Destino));

    strcpy(d->ciudad, ciudad);
    d->latitud = latitud;
    d->longitud = longitud;

    return d;
}

char* getCiudad(DestinoPtr d){
    return d->ciudad;
}
float getLatitud(DestinoPtr d){
    return d->latitud;
}
float getLongitud(DestinoPtr d){
    return d->longitud;
}

void setCiudad(DestinoPtr d, char* ciudad){
    strcpy(d->ciudad, ciudad);
}
void setLatitud(DestinoPtr d, float latitud){
    d->latitud = latitud;
}
void setLongitud(DestinoPtr d, float longitud){
    d->longitud = longitud;
}

void mostrarDestino(DestinoPtr d){
    printf("\n---------DESTINO---------\n");
    printf("\nCIUDAD: %s\n", d->ciudad);
    printf("\nLATITUD: %f\n", d->latitud);
    printf("\nLONGITUD: %f\n", d->longitud);
}

void liberarDestino(DestinoPtr d){
    free(d);
}

File: CLASE VII - Listas Void con Struct Anidadas/pre_parcial/golang/destino.go
package main

import "fmt"

type Destino struct {
	ciudad   string
	latitud  float32
	longitud float32
}

func crearDestino(ciudad string, latitud, longitud float32) *Destino {
	return &Destino{ciudad: ciudad, latitud: latitud, longitud: longitud}
}

func (d *Destino) GetCiudad() string           { return d.ciudad }
func (d *Destino) GetLatitud() float32         { return d.latitud }
func (d *Destino) GetLongitud() float32        { return d.longitud }
func (d *Destino) SetCiudad(c string)          { d.ciudad = c }
func (d *Destino) SetLatitud(lat float32)      { d.latitud = lat }
func (d *Destino) SetLongitud(lon float32)     { d.longitud = lon }

func mostrarDestino(d *Destino) {
	fmt.Printf("\n---------DESTINO---------\n")
	fmt.Printf("\nCIUDAD: %s\n", d.ciudad)
	fmt.Printf("\nLATITUD: %f\n", d.latitud)
	fmt.Printf("\nLONGITUD: %f\n", d.longitud)
}

